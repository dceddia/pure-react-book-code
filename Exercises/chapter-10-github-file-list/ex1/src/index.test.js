jest.mock('moment');
jest.mock('react-dom', () => ({render: jest.fn()}))

import React from 'react';
import { FileList, testFiles } from './index';
import renderer from 'react-test-renderer';
import moment from 'moment';

moment.mockImplementation(t => ({
  fromNow: () => "10 months ago"
}));

it('renders', () => {
  const tree = renderer.create(
    <FileList files={testFiles} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
