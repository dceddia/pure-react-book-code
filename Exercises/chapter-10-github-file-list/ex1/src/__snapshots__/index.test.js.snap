// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<table
  className="file-list"
>
  <tbody>
    <tr
      className="file-list-item"
    >
      <td
        className="file-icon"
      >
        <i
          className="fa fa-folder"
        />
      </td>
      <td
        className="file-name"
      >
        <span>
          src
        </span>
      </td>
      <td
        className="commit-message"
      >
        <span>
          Initial commit
        </span>
      </td>
      <td
        className="age"
      >
        <span
          className="time"
        >
          10 months ago
        </span>
      </td>
    </tr>
    <tr
      className="file-list-item"
    >
      <td
        className="file-icon"
      >
        <i
          className="fa fa-folder"
        />
      </td>
      <td
        className="file-name"
      >
        <span>
          tests
        </span>
      </td>
      <td
        className="commit-message"
      >
        <span>
          Initial commit
        </span>
      </td>
      <td
        className="age"
      >
        <span
          className="time"
        >
          10 months ago
        </span>
      </td>
    </tr>
    <tr
      className="file-list-item"
    >
      <td
        className="file-icon"
      >
        <i
          className="fa fa-file-text-o"
        />
      </td>
      <td
        className="file-name"
      >
        <span>
          README
        </span>
      </td>
      <td
        className="commit-message"
      >
        <span>
          Added a readme
        </span>
      </td>
      <td
        className="age"
      >
        <span
          className="time"
        >
          10 months ago
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
